src/lib.ts

```diff
@@ -267,3 +267,29 @@ export function renderSegments(segments: Segment<SegmentType>[]): string {
     }
   }).join('');
 }
+
+export function withCache(fetcher: typeof fetch = fetch): typeof fetch {
```

Comment:
Why not make a custom wrapped `fetch` function instead of importing and using `withCache` everywhere?